rio_tiler/io/cogeo.py

```diff
@@ -198,10 +195,24 @@ def _set_zooms(self):
             )
             minzoom, maxzoom = self.tms.minzoom, self.tms.maxzoom
 
-        self.minzoom = self.minzoom if self.minzoom is not None else minzoom
-        self.maxzoom = self.maxzoom if self.maxzoom is not None else maxzoom
+        self._minzoom = minzoom
+        self._maxzoom = maxzoom
         return
 
+    @property
+    def minzoom(self):
+        """Return dataset minzoom."""
+        if self._minzoom is None or self._maxzoom is None:
+            self._set_zooms()
```

Comment:
This is a problem in rio-tiler 3.0 too but I'm realizing that the `tilesize` argument accepted by `get_zooms` is not passed through which probably isn't a huge deal because 256 is most common, but wouldn't this break if using a TMS with a larger tile size (ex. 512)?

Maybe `tilesize` argument should be set from `self.tms`?

Reply:
yeah I'm not sure about this. it's kinda weird, basically a TMS can have a tilesize but the user can also set the tilesize to what ever it wants.

I think we should get the tilesize from the TMS tho.